pkg/scaffold/v2/makefile.go

```diff
@@ -111,17 +111,5 @@ docker-push:
 # find or download controller-gen
 # download controller-gen if necessary
 controller-gen:
-ifeq (, $(shell which controller-gen))
-	@{ \
-	set -e ;\
-	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
-	cd $$CONTROLLER_GEN_TMP_DIR ;\
-	go mod init tmp ;\
-	go get sigs.k8s.io/controller-tools/cmd/controller-gen@{{.ControllerToolsVersion}} ;\
-	rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
-	}
-CONTROLLER_GEN=$(GOBIN)/controller-gen
-else
-CONTROLLER_GEN=$(shell which controller-gen)
-endif
+	go get sigs.k8s.io/controller-tools@{{.ControllerToolsVersion}}
```

Comment:
I would keep `CONTROLLER_GEN=$(GOBIN)/controller-gen`.

Reply:
Thats absolutely possible, it would look like
`CONTROLLER_GEN="go run sigs.k8s.io/controller-tools/cmd/controller-gen"`

Now i kinda want to change it that way, the more we need to use it, the better it is as a variable

But we have to keep the go get part: From experience, if we don't strictly define it as dev-dependency (see lower part of this PR), a go mod tidy **will remove** it from go.mod

Reply:
@estroz The `$(GOBIN)` approach is less OS-friendly.

Reply:
i just pushed https://github.com/kubernetes-sigs/kubebuilder/pull/1289/commits/3bc20527b84716661b4aae02393f52908cc0c414, GOBIN is gone